packages/amplify-provider-awscloudformation/src/aws-utils/aws-amplify-backend.ts

```diff
@@ -0,0 +1,25 @@
+import AWS from 'aws-sdk';
+import aws from './aws';
+import { loadConfiguration } from '../configuration-manager';
+import { $TSContext } from 'amplify-cli-core';
+
+export class AmplifyBackend {
+  public amplifyBackend: AWS.AmplifyBackend;
+
+  constructor(private readonly context: $TSContext, options = {}) {
```

Comment:
to avoid the weirdness of using async in a constructor, I'd recommend restructuring this class to be an `AmplifyBackendClientFactory` with an async `getAmplifyBackendClient()` method.

Alternatively, I'm not sure this needs to be a class at all, could just be an async function

Reply:
This is the pattern I observed in other new AWS classes implement and went ahead with the same pattern. eg: aws-ecr.ts

Reply:
They might have a reason for doing it that way, but it doesn't seem like we need this additional complexity here. The immediately-invoked async function in a constructor is a bit unorthodox imo. If an async function / method can do what we need it to, why not use that?

Reply:
Discussed offline and changed.